lib/portage/dbapi/vartree.py

```diff
@@ -1677,16 +1678,16 @@ def root(self):
     def getpath(self, mykey, filename=None):
         return self.dbapi.getpath(mykey, filename=filename)
 
-    def zap(self, mycpv):
+    def zap(self, mycpv: str) -> None:
         return
 
-    def inject(self, mycpv):
+    def inject(self, mycpv: str) -> None:
         return
 
-    def get_provide(self, mycpv):
+    def get_provide(self, mycpv: str) -> list[Any]:
```

Comment:
Returning "Any" as a kind of TODO is bad practice. We should ask what this actually is.

BTW the type is "list[str]" prior to 2018 and typing.Never after that.

Reply:
In principle, I'm not against Any as a TODO given how rough the situation is with typing in Portage. What I am against however is comments not being replied to and marked as resolved, implying they've been addressed, not that the author just disagrees... :(

Reply:
Note that my specific reason for disliking Any as a TODO is that Any is a valid annotation for generics. You can't easily know whether it's intended as a TODO or whether the function is actually generic.

You don't need TODOs at all. A function lacking an annotation is already a "TODO: add some annotations here?"

What remains unclear to me is why it's more important to annotate functions with anything at all, as long as they are annotated, than to incrementally annotate what we know is annotatable today and leave what we can't annotate for tomorrow.